src/runtime/refiner.ts

```diff
@@ -750,3 +786,14 @@ class RefinementOperator {
     }
   }
 }
+
+export class SQLExtracter {
+  static fromSchema(schema: Schema, table: string): string {
+    let schemaFilter: string = schema.refinement && schema.refinement.toSQLExpression();
+    for (const fieldName of Object.keys(schema.fields)) {
+      const fieldFilter = schema.fields[fieldName].refinement && schema.fields[fieldName].refinement.toSQLExpression();
+      schemaFilter = schemaFilter ? schemaFilter + (fieldFilter ? (' AND ' + fieldFilter) : '') : fieldFilter;
+    }
+    return `SELECT * FROM ${table}` + (schemaFilter ? ` WHERE ${schemaFilter}` : '') + ';';
```

Comment:
Few things here:

1. You can iterate over object keys with `for (const fieldName in schema.fields))`
2. ... but you're actually just using the values, so you could:
```
for (const field of Object.values(schema.fields)) {
  const fieldFilter = field.refinement && field.refinement.toSQLExpression();
```
3. ... but I think this can be better expressed by building a list of terms and constructing the string in one go:

```
const filterTerms = [];
if (schema.refinement) {
  filterTerms.push(schema.refinement.toSQLExpression());
}
for (const field of Object.values(schema.fields)) {
  if (field.refinement) {
    filterTerms.push(field.refinement.toSQLExpression());
  }
}
return `SELECT * FROM ${table}` + (filterTerms ? ` WHERE ${filterTerms.join(' AND ')}` : '') + ';';
```

(At least, I think that'll work :) )

Reply:
I think,
> ... (filterTerms ? ` WHERE ...

will have to change to

> ... (filterTerms.length ? ` WHERE ...

and then it should work.
But this is a lot more readable. I'll update my code. Thanks Myk!

Reply:
Ah yes, I was thinking in python :P